frontend/src/components/ValueUnitField.tsx

```diff
@@ -59,26 +68,41 @@ const ValueUnitField: React.FC<ValueUnitFieldProps> = ({
       </SplitItem>
       <SplitItem>
         <Dropdown
-          data-testid="value-unit-select"
-          menuAppendTo="parent"
-          toggle={
-            <DropdownToggle id="toggle-basic" onToggle={() => setOpen(!open)}>
-              {currentUnitOption.name}
-            </DropdownToggle>
-          }
-          isOpen={open}
-          dropdownItems={options.map((option) => (
-            <DropdownItem
-              key={option.unit}
-              onClick={() => {
-                onChange(`${currentValue}${option.unit}`);
-                setOpen(false);
+          shouldFocusToggleOnSelect
+          toggle={(toggleRef) => (
+            <MenuToggle
+              data-testid="value-unit-select"
+              aria-label="value unit field toggle"
```

Comment:
Was there a previous value for `aria-label` that could be used here?

Reply:
no aria-label used before